src/plugins/intel_gpu/src/runtime/layout.cpp

```diff
@@ -505,6 +505,16 @@ bool layout::compatible(const layout& other) const {
     auto l1_pitch = l1.get_pitches();
     auto l2_pitch = l2.get_pitches();
 
+    // Ignore pitches which will never be used (for dims with size == 1)
```

Comment:
What is the target case? Can you please add related tests?

How about the following case? 
```
l1_size:   [2,  1,  2 (pad_after + 2), 4]
l1_pitch:  [16, 16, 4, 1]

l2_size:   [2,  1 (pad_after + 1), 2, 4]
l2_pitch:  [16, 8, 4, 1]
```
The sizes are the same, and if we zero out some pitches, will we treat them as identical?

Reply:
Added a couple of tests